build-tools/automation/azure-pipelines.yaml

```diff
@@ -67,6 +67,11 @@ variables:
   NUnit.NumberOfTestWorkers: 4
   GitHub.Token: $(github--pat--vs-mobiletools-engineering-service2)
   CONVERT_JAVADOC_TO_XMLDOC: $[ne(variables['Build.DefinitionName'], 'Xamarin.Android-PR')]
+  ${{ if eq(variables['Build.DefinitionName'], 'Xamarin.Android-PR') }}:
+    MicroBuildSignType: Test
+  ${{ if and(eq(variables['Build.DefinitionName'], 'Xamarin.Android'), ne(variables['Build.Reason'], 'PullRequest')) }}:
+    MicroBuildSignType: Real
```

Comment:
Is there a way to put an `else` to make sure `$(MicroBuildSignType)` is set to `Test` if it were blank?

No idea if this syntax works:
```yaml
 ${{ if and(eq(variables['Build.DefinitionName'], 'Xamarin.Android'), ne(variables['Build.Reason'], 'PullRequest')) }}:
    MicroBuildSignType: Real
 ${{ else }}
    MicroBuildSignType: Test
```

Reply:
I am fairly sure there's no way to do this inline in yaml, at least it's not documented anywhere that I can find:
https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#conditionally-assign-a-variable

Reply:
Should we put an `if` that has the expression inverted?
```yaml
 ${{ if and(eq(variables['Build.DefinitionName'], 'Xamarin.Android'), ne(variables['Build.Reason'], 'PullRequest')) }}:
    MicroBuildSignType: Real
 ${{ if or(ne(variables['Build.DefinitionName'], 'Xamarin.Android'), eq(variables['Build.Reason'], 'PullRequest')) }}:
    MicroBuildSignType: Test
```

Reply:
I've made this change, seems reasonable to make sure some value will be set here if we introduce any new pipelines using this file in the future.